sdk/trace/provider.go

```diff
@@ -210,42 +212,38 @@ func (p *TracerProvider) UnregisterSpanProcessor(sp SpanProcessor) {
 	if p.isShutdown.Load() {
 		return
 	}
-	old := *(p.spanProcessors.Load())
+	old := p.getSpanProcessors()
 	if len(old) == 0 {
 		return
 	}
-	spss := spanProcessorStates{}
-	spss = append(spss, old...)
+	spss := make(spanProcessorStates, len(old))
+	copy(spss, old)
 
 	// stop the span processor if it is started and remove it from the list
-	var stopOnce *spanProcessorState
-	var idx int
 	for i, sps := range spss {
-		if sps.sp == sp {
-			stopOnce = sps
-			idx = i
+		if sps.sp != sp {
```

Comment:
This algorithm changed. It used to unregister the last matching span processor, but now it unregisteres the first matching.

Should it be changed unregister all instances instead?

Reply:
You are correct. I wonder if the behavior should be specified in the spec?

Reply:
I've removed the commit with this change, but kept the tests.